### ISSUE TITLE
[settings] Add payment fees pill

### ISSUE THREAD

**Issue by cesarcosta99** [woocommerce/woocommerce-gateway-stripe#2091]
## Description

Add the payment fees pill to the payment methods list. Pill should be displayed during UPE onboarding flow and in the settings.

![image](https://user-images.githubusercontent.com/10233985/138365178-a1989949-c64d-406d-8ea8-cb9790c2643f.png)

## Questions

- Is there a reference for all the payment methods fees values?
- Do we need to provide a tooltip or something else to make the fee composition transparent to the merchant?

## Acceptance criteria

- UI matches design
- Fees are consistent in both settings and onboarding flow

**Comment by reykjalin** [comment]
@frosso I'm picking up this issue and see that the fees were hidden behind a feature flag in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1890. Can we just disable that feature flag? And what I mean by that are the fees correct as soon as we disable the flag or will they have to be corrected? 🤔

**Comment by reykjalin** [comment]
> And what I mean by that are the fees correct as soon as we disable the flag or will they have to be corrected? 🤔

🤦‍♂️  I completely missed this part of the code that makes this _very obvious_:

https://github.com/woocommerce/woocommerce-gateway-stripe/blob/5d841445e1361848ba0a22acdc25f80d1c2f2828/client/components/payment-method-fees-pill/index.js#L11-L13

So let me turn the question into this:

Can I just remove the following webpack config after the fees have been correctly fetched and displayed?

https://github.com/woocommerce/woocommerce-gateway-stripe/blob/5d841445e1361848ba0a22acdc25f80d1c2f2828/webpack.config.js#L20-L24

**Comment by reykjalin** [comment]
On retrieving the fees — I'm not sure how to actually accomplish that. I don't see anything in Stripe's API documentation that indicates that you can retrieve the fee associated with each payment method.

I'm not sure how feasible it is to implement this without manually specifying the fees Stripe charges for each transaction. I'll reach out to Stripe to ask about this 👍

**Comment by frosso** [comment]
> Can I just remove the following webpack config after the fees have been correctly fetched and displayed?

Yes, absolutely! I put it behind a feature flag within Webpack so that the code wouldn't be bundled when the flag was disabled (since, as you found, the fees are hardcoded).

> On retrieving the fees — I'm not sure how to accomplish that. I don't see anything in Stripe's API documentation that indicates that you can retrieve the fee associated with each payment method.

I think this is where we also got stuck.
The last time we contacted Stripe about it, there wasn't an endpoint that returns the fees. But maybe things have changed, now :)
Here's another ticket that might have some information (and seems to be duplicate to this?): https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1766
Some of the main issues on this are that:
- The fees for each country can be complicated - We don't know the exact fees until we make a transaction
- Some merchants have volume-based fees

I hope this helps a little bit!

**Comment by reykjalin** [comment]
That helps a lot @frosso, thank you!

**Comment by frosso** [comment]
Closing as not planned. This sat behind the `__PAYMENT_METHOD_FEES_ENABLED` flag since 2021 and we never flipped it on, and the blocker we hit back then still holds

I'm removing the dead flag-gated code as part of this. If Stripe ever ships a fees endpoint, we can always revisit 🙂
